engine/consensus/compliance/engine_test.go

```diff
@@ -219,15 +220,14 @@ func (cs *ComplianceSuite) TestOnFinalizedBlock() {
 	cs.head = &finalizedBlock
 
 	done := atomic.NewBool(false)
```

Comment:
same as [my comment above](url): would suggest to leaven concurrency handling to testify